src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1766,6 +1769,122 @@ export class SavedObjectsRepository {
     };
   }
 
+  async getPermissionQueryDSL(props: {
```

Comment:
Will these two new repository methods be used in other places? For now, they are only be called in the saved object client. Is that means we will call  `getPermissionQueryDSL` or  `processFindOptions` directly saved object client in the real case?

Reply:
Should we name it QueryDSL? If the data will not be stored in open search?

Reply:
> Will these two new repository methods be used in other places? For now, they are only be called in the saved object client. Is that means we will call `getPermissionQueryDSL` or `processFindOptions` directly saved object client in the real case?

There is no difference to call these two methods through savedObjectsClient or repositoryClient, do you want to say that we should define these two methods only in repositoryClient?

Reply:
> Should we name it QueryDSL? If the data will not be stored in open search?

I think it is OK to name it `QueryDSL` as it stands for "query domain specific language" here and it is not a opensearch specific term.